src/components/MoveAdder.ts

```diff
@@ -15,4 +15,241 @@ export class MoveAdder<TGameStartr extends FullScreenPokemon> extends GeneralCom
     public addMove(pokemon: IPokemon, move: IMove, index: number) {
         pokemon.moves[index] = move;
     }
+
+    /**
+     * Brings up the dialog for teaching a Pokemon a new move.
+     *
+     * @param pokemon   The pokemon whose moveset is being modified.
+     * @param move   The move that's going to be added into the moveset.
+     */
+    public startDialog(pokemon: IPokemon, move: IMove): void {
+        this.gameStarter.menuGrapher.deleteMenu("Yes/No");
+        this.gameStarter.menuGrapher.deleteMenu("GeneralText");
+
+        this.gameStarter.menuGrapher.createMenu("GeneralText");
+        if (this.alreadyKnowsMove(pokemon, move)) {
+            return;
+        }
+        if (pokemon.moves.length < 4) {
+            this.learnsNewMove(pokemon, move);
+        } else {
+            this.resolveMoveConflict(pokemon, move);
+        }
+    }
+
+    /**
+     * Checks if a Pokemon is trying to learn a move it already knows.
+     *
+     * @param pokemon   The pokemon whose moveset is being modified.
+     * @param move   The move that's going to be added into the moveset.
+     */
+    private alreadyKnowsMove(pokemon: IPokemon, move: IMove): boolean {
+        let counter = false;
+        for (const element of pokemon.moves) {
+            if (element.title === move.title) {
+                counter = true;
```

Comment:
Notice how the `counter` value only cares about the first time it's used. Put the dialog creation in here, and return immediately. That way you don't have to loop over the remaining moves.